File: backend/schemas/marketplace.py
from pydantic import BaseModel
from datetime import datetime


class MarketplaceItemRead(BaseModel):
    id: str
    name: str
    description: str
    type: str
    config: dict
    status: str
    submitted_by: str
    reviewed_by: str | None
    discovered_tools: list | None
    created_at: datetime
    model_config = {"from_attributes": True}


class PaginatedMarketplace(BaseModel):
    items: list[MarketplaceItemRead]
    total: int
    page: int
    limit: int

File: frontend/src/api/marketplace.ts
import client from './client';

export interface MarketplaceItem {
  id: string;
  name: string;
  description: string;
  type: 'mcp' | 'skill';
  config: any;
  file_path?: string;
  status: 'pending' | 'approved' | 'rejected';
  submitted_by: string;
  reviewed_by?: string;
  status_changed_at?: string;
  discovered_tools?: string[];
  created_at: string;
}

export const getMarketplaceItems = (type?: 'mcp' | 'skill') =>
  client.get<{items: MarketplaceItem[], total: number, page: number, limit: number}>('/marketplace', {
    params: type ? { type } : {}
  }).then(r => r.data);

export const getPendingItems = () =>
  client.get<{items: MarketplaceItem[], total: number, page: number, limit: number}>('/marketplace/pending').then(r => r.data);

export const submitMarketplaceItem = (formData: FormData) =>
  client.post<{id: string}>('/marketplace/submit', formData).then(r => r.data);

export const approveItem = (id: string) =>
  client.post(`/marketplace/${id}/approve`).then(r => r.data);

export const rejectItem = (id: string) =>
  client.post(`/marketplace/${id}/reject`).then(r => r.data);
